### PR TITLE
[Task 1872013] add default value for functions

### PR DESCRIPTION
**Pull request by andxu**
[AB#1872013](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1872013)

### CODE REVIEW

File: azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/DeployMojo.java [hunk 1]
```diff
@@ -135,9 +144,45 @@ protected void validateParameters() {
...
+        final boolean newFunctionApp = !app.exists();
+        AppServiceConfig defaultConfig = !newFunctionApp ? fromAppService(app, app.plan()) : buildDefaultConfig(config.subscriptionId(),
+            config.resourceGroup(), config.appName());
+
+        mergeAppServiceConfig(config, defaultConfig);
+        if (config.pricingTier() == null) {
+            config.pricingTier(PricingTier.CONSUMPTION);
+        }
+        try {
+            com.microsoft.azure.toolkit.lib.appservice.utils.Utils.mergeObjects(config, defaultConfig);
```
Comment (andxu) [inline azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/DeployMojo.java:157]: remove this statement

File: azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/DeployMojo.java [hunk 2]
```diff
@@ -135,9 +144,40 @@ protected void validateParameters() {
     }
 
     protected IFunctionAppBase<?> createOrUpdateResource(final FunctionAppConfig config) {
+        IFunctionApp app = Azure.az(AzureAppService.class).functionApp(config.resourceGroup(), config.appName());
+        final boolean newFunctionApp = !app.exists();
+        AppServiceConfig defaultConfig = !newFunctionApp ? fromAppService(app, app.plan()) : buildDefaultConfig(config.subscriptionId(),
+            config.resourceGroup(), config.appName());
+
+        mergeAppServiceConfig(config, defaultConfig);
+        if (config.pricingTier() == null) {
```
Comment (andxu) [inline azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/DeployMojo.java:153]: shall delete these lines:
```
if (config.pricingTier() == null) {
...
}
```

File: azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/DeployMojo.java [hunk 3]
```diff
@@ -135,9 +144,36 @@ protected void validateParameters() {
...
+            config.appInsightsKey(app.entity().getAppSettings().get(CreateOrUpdateFunctionAppTask.APPINSIGHTS_INSTRUMENTATION_KEY));
+        }
         return new CreateOrUpdateFunctionAppTask(config).execute();
     }
 
+    private AppServiceConfig buildDefaultConfig(String subscriptionId, String resourceGroup, String appName) {
+        ComparableVersion javaVersionForProject = null;
+        final String outputFileName = project.getBuild().getFinalName() + "." + project.getPackaging();
+        File outputFile = new File(project.getBuild().getDirectory(), outputFileName);
+        if (outputFile.exists() && StringUtils.equalsIgnoreCase("jar", FilenameUtils.getExtension(outputFile.getName()))) {
```
Comment (Flanker32) [inline azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/DeployMojo.java:163]: May remove the check for extension name as packaging for function should be jar

**Review by Flanker32 — COMMENTED** [review]
LGTM

File: azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/DeployMojo.java (hunk 3 above)
Comment (andxu) [reply azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/DeployMojo.java:163]: Minor change

**Review by Flanker32 — APPROVED** [review]

**Pull request merged**